### ISSUE TITLE
Astropy tutorials on Docker / tempnb?

### ISSUE THREAD

**Issue by cdeil** [astropy/astropy-tutorials#74]
With [Docker](https://www.docker.com/) and [tempnb](https://github.com/jupyter/tmpnb) it should be possible to make the Astropy tutorials available online ... any user can run the notebooks from any computer without needing to install things locally.

Check out  https://tempnb.org for a demo and https://lambdaops.com/ipythonjupyter-tmpnb-debuts/  for some more info on how it works.

Here's what needs to happen:
- [x] Make a free account for the Astropy organization here: https://registry.hub.docker.com/search?q=astropy (@astrofrog or @eteq: could you do this and give me access?)
- [ ] Make one or a few Docker containers (e.g. one with Astropy only, then one with the prominent affiliated packages installed, and one with the tutorial included) and put them on the docker hub (I'm interested in setting this up and maintaining it.)
- [ ] Find someone willing to deploy this or give some money so that we can deploy this in some cloud (maybe we could ask Numfocus or Rackspace).

This could be nice to have in the Astropy 1.0 release announcement.

**Comment by cdeil** [comment]
Here's a first version of a Docker file and image that contains Astropy:
https://github.com/gammapy/gammapy/tree/master/dev/docker

**Comment by adrn** [comment]
Sorry for dropping the ball on this, @cdeil , and thanks for making me aware of this!

**Comment by eteq** [comment]
@cdeil - I know this is months old, but I just noticed this and did this now. I created a [docker organization for Astropy](https://registry.hub.docker.com/repos/astropy/) and gave you full access to it.

**Comment by cdeil** [comment]
@eteq – Thanks for setting this up.

I'm still very interested to learn more about Docker and set this up, but realistically I'm swamped with other things until July. So if someone else wants to do this, please go ahead.

These could serve as examples / templates .. maybe all that's needed is to change the Dockerfile to install Astropy and the tutorial repo:
* https://github.com/jupyter/docker-demo-images/
* https://github.com/scikit-image/skimage-sandbox

**Comment by eteq** [comment]
Same here re: swamped, but I do agree a docker template would be a valuable thing to have in place.  If anyone else wants to try this (maybe @adrn?) and needs access, feel free to ping me.

**Comment by cdeil** [comment]
As mentioned at https://github.com/x-calibre/astropython/issues/100, it would be great if we can get online IPython notebooks for the Astropy tutorials or other Astro packages going, either included in astropython.org somehow, or linked to from that page and the Astropy page / docs.

So I asked @rgbkrk from Jupyter / Rackspace for advice on this via email with @x-calibre in CC, but now I think it's better continue the discussion here so that @eteq, @adrn, @taldcroft or anyone else interested in this can comment or help make this happen.

To summarise the advice from @rgbkrk from the email:
* At this time,  tempnb deployment (like https://try.jupyter.org) is not easy. Only attempt it if there's a sysadmin that will maintain it. An example is http://notebooks.codeneuro.org/ and their setup is at https://github.com/codeneuro/notebooks . They are working on simplifying this, with the goal that the end user simply involves clicking a button on a github repo and spawning a notebook setup just for them. 
* It might be possible to include the astro packages and example notebooks in http://try.jupyter.org/, see https://github.com/jupyter/docker-demo-images/blob/master/Dockerfile#L109-L110 as an example. This would of course be wonderful because it allows us to focus on the content, instead of on the deployment. I don't know if / how much we'd have to pay, obviously there is a significant cost to run / maintain this and maintain it (e.g. updates to our packages and notebooks).
* One issue is that these notebooks don't have internet access, so all the small example data files have to be bundled in the repo and docker container instead of being downloaded at the start of the notebook (see http://www.astropy.org/astropy-tutorials/FITS-images.html#Viewing-and-manipulating-FITS-images). @adrn Is this a change you feel comfortable with or is it important to not bundle the example data in the repo for some reason?

**Comment by taldcroft** [comment]
Support for this for generic astropy-related notebooks would be amazing indeed!  For starters this would provide motivation to finally migrate python4astronomers over to Jupyter notebooks so they could then be run live by students in tutorial sessions with no installation.

**Comment by cdeil** [comment]
@rgbkrk commented "There's no cost with adding your stuff to docker-demo-images. It's community run and the only current sponsor is Rackspace."

So if @x-calibre or anyone else has time to work on this (adding Astropy stuff to `docker-demo-images`) in the coming weeks, please leave a comment here. Otherwise I'll give it a try next week.

**Comment by x-calibre** [comment]
I have very limited experience with Docker but would love to learn and apply it here ... I would be available in about 2 weeks, then I can definitely give this a try...

**Comment by adrn** [comment]
Here's another possibility: http://mybinder.org/

**Comment by x-calibre** [comment]
@cdeil  @adrn @eteq @taldcroft  
I have put up a very limited demo version of this on http://www.astropython.org/interactive
If this is what we are aiming for , then I can build a creation form for users to upload/link existing IPython Notebooks along with the list of requirements or the docker file on the website and add it to the list.

The only background task that needs to be done is to provide a link to the notebook from the admin panel (after feeding it to the Binder application)

**Comment by cdeil** [comment]
We now set up Binder for Gammapy: https://gammapy.readthedocs.io/en/latest/getting-started.html#binder

I see that binder is also set up for this repo:
https://github.com/astropy/astropy-tutorials#running-tutorials
(modulo #110).

Binder seems like the simplest solution to set up and maintain.
Closing this issue now so that we don't have to read through the discussion.
If there's something to be done, let's open new issues or PRs.
